server.go

```diff
@@ -103,21 +103,22 @@ type Response struct {
 	ErrType responseErr
 }
 
-// Permissions maps "service.method" of a peer to boolean permission
-type Permissions map[peer.ID]map[string]bool
-
-// AuthorizeWithMap returns an authrorization function that follows the authorization strategy as described in the
+// AuthorizeWithMap returns an authrorization function that follows the strategy as described in the
 // given map(maps "service.method" of a peer to boolean permission)
-func AuthorizeWithMap(p Permissions) func(peer.ID, string, string) bool {
-	if p != nil {
-		return func(pid peer.ID, svc string, method string) bool {
-			return p[pid][svc+"."+method]
+func AuthorizeWithMap(p map[peer.ID]map[string]bool) func(peer.ID, string, string) bool {
+	return func(pid peer.ID, svc string, method string) bool {
+		// If map is nil, no method would be allowed
+		if p == nil {
+			return false
 		}
+		return p[pid][svc+"."+method]
 	}
+}
 
-	// If permission map is nil, no method would be allowed
-	return func(pid peer.ID, svc string, method string) bool {
-		return false
+// WithAuthorize adds authorization to the server using given authorization function
+func WithAuthorize(a func(peer.ID, string, string) bool) ServerOption {
```

Comment:
I don't know if it's allowed, but `WithAuthorize(a func(pid peer.ID, name string, nethod string) bool) ServerOption {`

will give the user better hits when reading the documentation about how that function will be called.

Also, can be additionally explained in writing in the godoc string. `WithAuthorizeFunc` might be a better name too.

Reply:
> I don't know if it's allowed, but `WithAuthorize(a func(pid peer.ID, name string, method string) bool) ServerOption {`

Should be. You can get away with having argument names in interface and having different argument names in the implementation.